### DISCUSSION TITLE
force update childproject package

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#255] [Q&A] [answered]
I'm trying to use the new and shiny metrics function documented [here](https://childproject.readthedocs.io/en/latest/metrics.html), but I'm getting this error:

` child-project metrics input/solomon solomon.csv aclew``

> usage: child-project [-h]
                     {validate,import-annotations,merge-annotations,intersect-annotations,remove-annotations,rename-annotations,import-data,overview,compute-durations,process,sampler,zooniverse,eaf-builder,anonymize}
                     ...
child-project: error: invalid choice: 'metrics' (choose from 'validate', 'import-annotations', 'merge-annotations', 'intersect-annotations', 'remove-annotations', 'rename-annotations', 'import-data', 'overview', 'compute-durations', 'process', 'sampler', 'zooniverse', 'eaf-builder', 'anonymize'

I think this means that I don't have the latest version of childproject. I tried to get it with:
`pip3 install git+https://github.com/LAAC-LSCP/ChildProject.git``

Full output is attached, but in a nutshell, it seems to be going through the right motion at the beginning:

> Collecting git+https://github.com/LAAC-LSCP/ChildProject.git
  Cloning https://github.com/LAAC-LSCP/ChildProject.git to /private/var/folders/mx/f_gpzmq105x8sgqfdtbr4zrr0000gp/T/pip-req-build-15aekuu7
  Running command git clone -q https://github.com/LAAC-LSCP/ChildProject.git /private/var/folders/mx/f_gpzmq105x8sgqfdtbr4zrr0000gp/T/pip-req-build-15aekuu7

followed with a bunch of "Requirement already satisfied:". I also checked to make sure I had the latest version of pip.

I feel like I've had this problem before, and I fixed it by uninstalling and re-installing childproject. I tried to do this:

``` 
rm -rf /private/var/folders/mx/f_gpzmq105x8sgqfdtbr4zrr0000gp/T/pip-req-build-15aekuu7
pip3 install git+https://github.com/LAAC-LSCP/ChildProject.git
```

But I still have the same error.

Output
[output-install.txt](https://github.com/LAAC-LSCP/ChildProject/files/6799444/output-install.txt)

**Comment by lucasgautheron** [comment, marked as answer]
Hello,

The solution is to use the `--upgrade` switch:

```
pip3 install git+https://github.com/LAAC-LSCP/ChildProject.git --upgrade
```

If that does not work, you may uninstall the package (`pip uninstall ChildProject`) then re-install it

**Reply by alecristia** [reply]
the upgrade switch still didn't do it for me, but this did:

```
pip uninstall ChildProject
pip3 install git+https://github.com/LAAC-LSCP/ChildProject.git --upgrade
```

**Reply by lucasgautheron** [reply]
Alright, not sure why --upgrade does not behave consistently.

Anyway, the option that worked for you is the one [documented in the docs](https://childproject.readthedocs.io/en/latest/install.html), even though it may be hard to find.

**Reply by alecristia** [reply]
you're right!! I see it in the note at the bottom of the installation page. And interestingly, I remembered we had come across this problem so I looked at the docs and didn't see it. I looked at the installation section (but not all the way down to the note) & the cheat sheet (and did a search for "update", not "upgrade", in the issues & discussions). Perhaps we should add a section on upgrades? or simply add this to the cheat sheet?
